test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5981,15 +5981,31 @@ def submit(self, obj, script=None, extend=None, submit_dir=None,
             os.chdir(submit_dir)
         c = None
         # 1- Submission using the command line tools
+        runcmd = []
+        exp_str = ""
+        if len(env_set) > 0:
+            if '()' in list(env_set.keys())[0]:
+                func = list(env_set.keys())[0]
+                f_body = list(env_set.values())[0]
+                f_name = func.replace('()', '')
+                runcmd += [func, f_body, "\n", "export", "-f", f_name]
+            else:
+                exp_str = "export "
```

Comment:
You don't need export.  You can say 'VAR=val ./a.out' and VAR will be set as an environmental variable.

Reply:
Hi Bhroam,

I tried the following as: 
#!/bin/bash
NONPRINT_VAR="X^AY"
/opt/pbs/bin/qsub /tmp/PtlPbsJobScriptbpfw37yc

Environment variable NONPRINT_VAR was not seen in qstat -f output.
I could see the environment variable only when i add export NONPRINT_VAR="X^AY"

Reply:
Yes, it won't show up if you create a script.  We had previously talked about not using a script and adding the variables directly to the command line.  VAR=val ./a.out is exported as an environmental variable.